storage/src/main/java/zipkin2/storage/kafka/KafkaAutocompleteTags.java

```diff
@@ -34,62 +42,105 @@
+    return new GetTagKeysCall(traceStoreStream);
   }
 
   @Override public Call<List<String>> getValues(String key) {
-    ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore =
-        traceStoreStream.store(AUTOCOMPLETE_TAGS_STORE_NAME,
-            QueryableStoreTypes.keyValueStore());
-    return new GetValuesCall(autocompleteTagsStore, key);
+    return new GetTagValuesCall(traceStoreStream, key);
   }
 
-  static class GetKeysCall extends KafkaStreamsStoreCall<List<String>> {
-    final ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore;
+  static class GetTagKeysCall extends Call.Base<List<String>> {
+    static final Gson GSON = new Gson();
+
+    final KafkaStreams traceStoreStream;
 
-    GetKeysCall(ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore) {
-      this.autocompleteTagsStore = autocompleteTagsStore;
+    GetTagKeysCall(KafkaStreams traceStoreStream) {
+      this.traceStoreStream = traceStoreStream;
     }
 
-    @Override protected List<String> query() {
-      List<String> keys = new ArrayList<>();
-      autocompleteTagsStore.all().forEachRemaining(keyValue -> keys.add(keyValue.key));
-      return keys;
+    @Override protected List<String> doExecute() throws IOException {
+      return traceStoreStream.allMetadataForStore(AUTOCOMPLETE_TAGS_STORE_NAME)
+          .parallelStream()
+          .map(KafkaAutocompleteTags::httpClient)
+          .map(httpClient -> {
+            AggregatedHttpResponse response = httpClient.get("/autocomplete-tags")
+                .aggregate()
+                .join();
+            if (!HttpStatus.OK.equals(response.status())) return null;
+            return response.contentUtf8();
+          })
+          .map(content -> {
+            Set<String> set = GSON.fromJson(content, Set.class);
+            return set;
+          })
+          .flatMap(Collection::stream)
+          .distinct()
+          .collect(Collectors.toList());
+    }
+
+    @Override protected void doEnqueue(Callback<List<String>> callback) {
+      try {
+        callback.onSuccess(doExecute());
+      } catch (IOException e) {
+        callback.onError(e);
+      }
     }
 
     @Override public Call<List<String>> clone() {
-      return new GetKeysCall(autocompleteTagsStore);
+      return new GetTagKeysCall(traceStoreStream);
     }
   }
 
-  static class GetValuesCall extends KafkaStreamsStoreCall<List<String>> {
-    final ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore;
-    final String key;
+  static class GetTagValuesCall extends Call.Base<List<String>> {
+    static final Gson GSON = new Gson();
+
+    final KafkaStreams traceStoreStream;
+    final String tagKey;
 
-    GetValuesCall(
-        ReadOnlyKeyValueStore<String, Set<String>> autocompleteTagsStore, String key) {
-      this.autocompleteTagsStore = autocompleteTagsStore;
-      this.key = key;
+    GetTagValuesCall(KafkaStreams traceStoreStream, String tagKey) {
```

Comment:
add javadoc that this retrieving values is not scatter gather as the other command is

Reply:
added into abstract class for single instance calls